### PR TITLE
Integration tile for TaskCall

### PR DESCRIPTION
**Pull request by riasat-ullah**
### What does this PR do?

We have added a tile for our integration - TaskCall.

### Motivation

We want to list our integration in the Datadog marketplace.

### CODE REVIEW

**Comment by urseberry** [comment]
Documentation review on https://datadoghq.atlassian.net/browse/DOCS-5874

**Comment by riasat-ullah** [comment]
Hi Ursula,

It doesn't seem like we have access to the file on Jira. Can you please
give us access to it?

- Riasat

Riasat Ullah

TaskCall

Web: www.taskcallapp.com <https://taskcallapp.com/>

(ES) +34 666-320-884

(US) +1 917-524-9404

File: taskcall/README.md
Comment (anawishnoff) [inline taskcall/README.md:7]:
```suggestion
TaskCall’s integration allows Datadog users to bolster their operations by improving their incident awareness and simplifying the process in which they are handled. Incidents sync up bi-directionally between the two platforms. Once Datadog alerts are received in TaskCall, users will be able to systematically manage all incidents without being interrupted. Users will also benefit from improved impact visibility from dependency graphs and status dashboards. With better understanding of the state of your overall infrastructure efficient resolutions can be achieved.
```

File: taskcall/README.md
Comment (anawishnoff) [inline taskcall/README.md:15]:
```suggestion
- Incidents are automatically resolved in TaskCall when alert conditions are no longer present.
```

File: taskcall/README.md
Comment (anawishnoff) [inline taskcall/README.md:21]: Please add the instructions directly in this file, rather than linking to an external page. Additionally, please add an Uninstallation section as described [here](https://docs.datadoghq.com/developers/integrations/oauth_for_integrations/#publish-the-oauth-client).

File: taskcall/README.md
Comment (anawishnoff) [inline taskcall/README.md:26]: This should be moved to Setup as a prerequisite

File: taskcall/manifest.json
Comment (anawishnoff) [inline taskcall/manifest.json:11]:
```suggestion
    "description": "Monitor and centralize Datadog incidents with TaskCall",
```

File: taskcall/manifest.json
Comment (anawishnoff) [inline taskcall/manifest.json:41]:
```suggestion
      "Offering::Integration",
```

File: taskcall/manifest.json
Comment (anawishnoff) [inline taskcall/manifest.json:44]: Please also specify the types of data you will be sending into Datadog and pulling out of Datadog, i.e.:

`Submitted Data Type::Metrics`
`Queried Data Type::Events`

**Review by anawishnoff — CHANGES_REQUESTED** [review]
Initial review from the product side. Let me know if there are any questions, thanks!

File: taskcall/manifest.json
Comment (bgoldberg122) [inline taskcall/manifest.json:9]: Note: once you add the uninstallation instructions to the readme, you'll need to add the following line. 
```suggestion
    "configuration": "README.md#Setup",
    "uninstallation": "README.md#Uninstallation",
    "support": "README.md#Support",
```

File: taskcall/manifest.json
Comment (bgoldberg122) [inline taskcall/manifest.json:63]: Is your integration sending in events? 

If not, because this is an API-based integration, you can remove the integration definition here and also delete the `service_checks.json` and `metadata.csv` files
```suggestion
  "assets": {},
```

**Review by bgoldberg122 — CHANGES_REQUESTED** [review]
Hi @riasat-ullah, thanks for opening this PR, this is looking good! Please also add `CODEOWNER` entries under `# Community partners` and `# Community Partners + Documentation`. You can copy `ably`'s entries in both places and replace their email address with yours.

File: taskcall/manifest.json
Comment (riasat-ullah) [reply taskcall/manifest.json]: Change corrected.

**Review by bgoldberg122 — CHANGES_REQUESTED** [review]
Thanks for addressing my comments! There's just one piece that was missed in the manifest update and a character fix.

**Comment by riasat-ullah** [comment]
Thanks for pointing that out. The suggested changes have been applied.

Riasat Ullah

TaskCall

Web: www.taskcallapp.com <https://taskcallapp.com/>

(ES) +34 666-320-884

(US) +1 917-524-9404

File: taskcall/manifest.json
Comment (anawishnoff) [inline taskcall/manifest.json:47]: Hi @riasat-ullah, I had suggested these as pure examples to show what the formatting of these classifier tags look like. Can you please confirm that the integration is sending metrics into Datadog and querying events out of Datadog? If not, can you please update accordingly? Thanks

File: taskcall/README.md
Comment (anawishnoff) [inline taskcall/README.md:24]: Please include a step for navigating to the Integration tile, going to the configure tab, and clicking Connect Accounts to kick off the OAuth flow.

File: taskcall/README.md
Comment (anawishnoff) [inline taskcall/README.md:48]: Please add a step that the user should navigate to the Datadog tile and click uninstall

**Review by anawishnoff — CHANGES_REQUESTED** [review]
A few more requested changes from my end

**Review by bgoldberg122 — CHANGES_REQUESTED** [review]
Thanks for addressing my comments. Your tile looks good to me.

One thing I noticed on your OAuth client though is that the `onboarding_url` links to a docs page. Could you instead please set the `onboarding_url` for your client to be the actual page where the customer would connect the accounts (i.e. the `Integrations tab on the Service details page` from your docs)? Your configuration steps will show up for the customers on the tile within Datadog, so it is more convenient for them to automatically be redirected to the page that actually lets them connect the accounts.

Once you've made the update, you can republish the client, so we have the most up-to-date version. **NOTE**: the same published client is still used, so make sure you keep the original published client's `id` and `secret`.

Please let me know if you have any questions.

File: taskcall/README.md
Comment (bgoldberg122) [inline taskcall/README.md:44]: Ah, this actually has to be an h2 or else it won't render correctly in the tile
```suggestion
## Uninstallation
```

File: taskcall/manifest.json
Comment (riasat-ullah) [reply taskcall/manifest.json:47]: Oh! I did not realize that. We are not submitting any metrics or querying from Datadog. We only update the incidents that are sent to us by webhooks. As per @bgoldberg122's suggestion we removed the assets section, metadata.csv and service_checks.json files. I am assuming that with those out of the way, we can do without the Submitted Data Type and Queried Data Type. I will remove them.

**Comment by riasat-ullah** [comment]
@bgoldberg122, we were using the second OAuth flow where the authentication process is initiated from a third party application (in this case from TaskCall). In that flow we did not need the onboarding URL. That's why we just pointed it to the integration guide on our site.

However, after @anawishnoff insisted that we implement the primary flow where the authentication is initiated from the integration tile, we have now made the necessary changes. We have restricted the flow to only be initiated from Datadog.

For context: We chose against the primary flow at first because like Datadog we also serve clients from multiple independent data centers. Adopting that flow (which we now have) with a single onboarding URL meant additional work to handle switching between the regions to make sure the flow is pointed to the correct region where the client's account is in.

File: taskcall/README.md
Comment (anawishnoff) [inline taskcall/README.md:27]:
```suggestion
3. On the TaskCall integration tile, go to the **Configure** tab and click **Connect Accounts**. You will be redirected to TaskCall.
```

File: taskcall/README.md
Comment (anawishnoff) [inline taskcall/README.md:32]:
```suggestion
5. Click **Integrate**. You will be redirected to a Datadog authorization page.
```

File: taskcall/README.md
Comment (anawishnoff) [inline taskcall/README.md:33]:
```suggestion
6. Authorize the integration, ensuring that you have the correct permissions to do so.
```

File: taskcall/README.md
Comment (anawishnoff) [inline taskcall/README.md:98]:
```suggestion
- Navigate to the [TaskCall integration tile][6] in Datadog and uninstall it. You must also delete the webhook you created to send notifications to TaskCall.
```

File: taskcall/README.md
Comment (anawishnoff) [inline taskcall/README.md:112]:
```suggestion
[5]: https://taskcallapp.com/images/vendors/datadog/DatadogTaskCallAuthorization.png
[6]: https://app.datadoghq.com/integrations/taskcall
```

File: taskcall/README.md
Comment (anawishnoff) [inline taskcall/README.md:26]:
```suggestion
2. **In Datadog**: Navigate to the [TaskCall integration tile][6].
```

File: taskcall/README.md
Comment (anawishnoff) [inline taskcall/README.md:100]: Since this integration does not create API keys, we can remove this line. 
```suggestion
```

**Review by anawishnoff — CHANGES_REQUESTED** [review]
A few minor comments on my end

File: taskcall/README.md
Comment (alai97) [inline taskcall/README.md:112]:
```suggestion
[6]: https://app.datadoghq.com/integrations/taskcall
[7]: https://app.datadoghq.com/integrations
```

File: taskcall/README.md
Comment (alai97) [inline taskcall/README.md:39]:
```suggestion
1. Go to [**Integrations** > **Integrations**][7].
```

File: taskcall/README.md
Comment (alai97) [inline taskcall/README.md:93]:
```suggestion
For more information, see the [TaskCall Datadog Integration Guide][3].
```

**Review by alai97 — APPROVED** [review]
Looks good, small suggestions!

**Review by brett0000FF — APPROVED** [review]
Looks good!

Before merging, you just need to move this image: https://taskcallapp.com/images/vendors/datadog/DatadogTaskCallAuthorization.png to the `/taskcall/images/` folder and update the reference link [5] to https://raw.githubusercontent.com/DataDog/integrations-extras/master/taskcall/images/DatadogTaskCallAuthorization.png

File: taskcall/README.md
Comment (bgoldberg122) [inline taskcall/README.md:113]: Looks like this instruction was accidentally removed (I also added the link to the api keys page).

```suggestion
- Additionally, ensure that all API keys associated with this integration have been disabled by searching for the integration name on the [API Keys page][8].


## Support

[Contact TaskCall support][4] for any questions regarding the integration or the platform.


[1]: https://app.us.taskcallapp.com/register
[2]: https://docs.taskcallapp.com/integrations/v1/datadog-integration-guide#in-datadog
[3]: https://docs.taskcallapp.com/integrations/v1/datadog-integration-guide
[4]: https://www.taskcallapp.com/contact-us
[5]: https://raw.githubusercontent.com/DataDog/integrations-extras/master/taskcall/images/DatadogTaskCallAuthorization.png
[6]: https://app.datadoghq.com/integrations/taskcall
[7]: https://app.datadoghq.com/integrations
[8]: https://app.datadoghq.com/organization-settings/api-keys?filter=TaskCall
```

**Review by bgoldberg122 — CHANGES_REQUESTED** [review]

File: taskcall/README.md
Comment (riasat-ullah) [reply taskcall/README.md:113]: We were asked to remove this line by @anawishnoff since we are not using any API keys.

Comment (bgoldberg122) [reply taskcall/README.md:113]: Oh, got it. Sorry, I didn't realize you weren't using that scope. Thanks for clarifying :) 

I'll go approve!

**Review by bgoldberg122 — APPROVED** [review]

File: taskcall/manifest.json
Comment (eho1307) [inline taskcall/manifest.json:45]:
```suggestion
      "Supported OS::macOS",
      "Queried Data Type::Incidents"
```

Comment (eho1307) [reply taskcall/manifest.json:45]: Jumping in here for Ana while she's out of office. Just adding one last tag to clarify that this pulls out incidents. This is the last change and then we can publish this to your sandbox.

**Review by eho1307 — APPROVED** [review]

**Pull request merged**
